src/main/java/campus/infrastructure/TaskList.java

```diff
@@ -0,0 +1,133 @@
+package campus.infrastructure;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import campus.exceptions.CampusException;
+import campus.tasks.Deadline;
+import campus.tasks.Event;
+import campus.tasks.Task;
+import campus.tasks.ToDos;
+
+/**
+ * TaskList Class handles all CRUD actions related to the Task Class, it also contains the List of 'Task'
+ * type field to store the various tasks in the list and does List Manipulation operations on them
+ */
+public class TaskList {
+    private List<Task> tasks;
+
+    public TaskList() {
+        this.tasks = new ArrayList<>();
+    }
+
+    /**
+     * Updates the current List of Task type given a List of String type input read from the saved txt file in memory
+     * @param listOfStrings List of String format
+     * @throws CampusException In the event that the file is corrupted and the List of String type does not match the
+     *      accepted txt file formatting for its data
+     */
+    public void updateListFromFile(List<String> listOfStrings) {
+        if (listOfStrings == null) {
+            return;
+        }
+
+        this.tasks = listOfStrings.stream()
+                .map(this::parseTaskFromStringSafe)
+                .filter(Objects::nonNull)
+                .collect(Collectors.toList());
+    }
+
+    private Task parseTaskFromStringSafe(String string) {
```

Comment:
naming is alittle long and not very clear

Reply:
yea...I think naming could be improved too, would you have any suggestions?